### PR TITLE
TODOS and MultiQC in DSL2 template

### PR DESCRIPTION
**Pull request by ggabernet**
Many thanks for contributing to nf-core/tools!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [ ] This comment contains a description of changes (with reason)
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [ ] Documentation in `docs` is updated
 - [ ] `CHANGELOG.md` is updated
 - [ ] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/tools/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by ggabernet** [comment]
@drpatelh Ok I had fixed it in a commit before the unwanted merge, so it should be ok:
https://github.com/nf-core/tools/blob/002fdac78f766ee9161fde01178d965d167268ae/nf_core/pipeline-template/%7B%7Bcookiecutter.name_noslash%7D%7D/bin/check_samplesheet.py#L87

Still don't know how it happened though, so sorry about that!
I'll continue in this PR now :)

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/bin/check_samplesheet.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #!/usr/bin/env python
 # TODO nf-core: Update the script to check the samplesheet
+# This script is based on the example at: https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/samplesheet/samplesheet_test_sispa.csv
```
Comment (ggabernet) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/bin/check_samplesheet.py:3]: I've added the link here to the example. Just thinking about it, adding the example in assets might be confusing and might make people think that the sample sheet for the test data should be placed there, maybe.

**Comment by ggabernet** [comment]
I'm not sure why exactly the launch tests are failing:
```
raise AttributeError(
>               "%s does not have the attribute %r" % (target, name)
            )
E           AttributeError: <module 'PyInquirer' from '/opt/hostedtoolcache/Python/3.6.10/x64/lib/python3.6/site-packages/PyInquirer/__init__.py'> does not have the attribute 'prompt'
```

**Comment by ewels** [comment]
This has been fixed upstream, will pull in updates: https://github.com/nf-core/tools/pull/668

**Comment by ewels** [comment]
@ggabernet - did you mean to add those two merge commits there? I had just fixed the Python Black linting but you reverted my fix 😅

**Comment by ewels** [comment]
@ggabernet - I force-pushed to remove your two merge commits. Please just run `git pull` on your fork locally and you will pull in my changes and be up to date 🚀

**Comment by ewels** [comment]
Tests passing ✅

**Comment by ggabernet** [comment]
Ah sorry I thought I was just pulling your changes

**Comment by ewels** [comment]
No problem :) Is this PR ready to merge now?

**Comment by ggabernet** [comment]
yes, the MultiQC part is working. Still fighting with the workflow summary for the MultiQC report but I can make another PR for that :)

**Comment by ggabernet** [comment]
This is now ready for review, I've added the workflow summary params into the MultiQC report as well :)

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/bin/check_samplesheet.py (hunk 1 above)
Comment (drpatelh) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/bin/check_samplesheet.py:3]: Thanks!

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/bin/check_samplesheet.py [hunk 2]
```diff
@@ -31,7 +32,14 @@ def print_error(error, line):
     sys.exit(1)
 
 
+# TODO nf-core: Update the check_samplesheet function
 def check_samplesheet(file_in, file_out):
+    """
+    This function checks that the sample sheet follows the following structure:
+    
+    sample,fastq_1,fastq_2
+    sample1,Sample1.fastq.gz,Sample2.fastq.gz
```
Comment (drpatelh) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/bin/check_samplesheet.py:41]:
```suggestion
    SAMPLE1,SAMPLE1_1.fastq.gz,SAMPLE1_2.fastq.gz
```

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf [hunk 3]
```diff
@@ -25,7 +25,7 @@ if (params.help) {
 /*
  * Stage config files
  */
-ch_multiqc_config = file("$baseDir/assets/multiqc_config.yaml", checkIfExists: true)
+ch_multiqc_config = Channel.fromPath("$baseDir/assets/multiqc_config.yaml", checkIfExists: true)
```
Comment (drpatelh) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf:28]: Why did you have to change this? Staging issues?

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf [hunk 4]
```diff
@@ -67,6 +67,9 @@ log.info Headers.nf_core(workflow, params.monochrome_logs)
 log.info summary.collect { k,v -> "${k.padRight(20)}: $v" }.join("\n")
 log.info "-\033[2m----------------------------------------------------\033[0m-"
 
+workflow_summary = Schema.params_mqc_summary(summary)
+ch_workflow_summary = Channel.value(workflow_summary)
```
Comment (drpatelh) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf:71]: Nice!

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/local/get_software_versions.nf [hunk 5]
```diff
@@ -9,8 +9,8 @@ process GET_SOFTWARE_VERSIONS {
                 }
 
     output:
-    path 'software_versions_mqc.yaml'
-    path "software_versions.csv"
+    path 'software_versions_mqc.yaml', emit: software_versions_yml
```
Comment (drpatelh) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/local/get_software_versions.nf:12]:
```suggestion
    path 'software_versions_mqc.yaml', emit: yml
```

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/local/get_software_versions.nf [hunk 6]
```diff
@@ -9,8 +9,8 @@ process GET_SOFTWARE_VERSIONS {
...
+    path "software_versions.csv", emit: software_versions_csv
```
Comment (drpatelh) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/local/get_software_versions.nf:13]:
```suggestion
    path "software_versions.csv", emit: csv
```

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf [hunk 7]
```diff
@@ -98,11 +101,12 @@ workflow {
...
-    // MULTIQC(
-    //     summary,
-    //     FASTQC.out,
-    //     ch_multiqc_config
-    // )
+    MULTIQC(
+        ch_multiqc_config.collect(),
+        ch_multiqc_custom_config.collect().ifEmpty([]),
+        FASTQC.out.collect(),
+        GET_SOFTWARE_VERSIONS.out.software_versions_yml.collect(),
```
Comment (drpatelh) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf:108]:
```suggestion
        GET_SOFTWARE_VERSIONS.out.yml.collect(),
```

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf [hunk 8]
```diff
@@ -98,11 +101,12 @@ workflow {
 
     GET_SOFTWARE_VERSIONS()
 
-    // MULTIQC(
-    //     summary,
-    //     FASTQC.out,
-    //     ch_multiqc_config
-    // )
+    MULTIQC(
+        ch_multiqc_config.collect(),
```
Comment (drpatelh) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf:105]:
```suggestion
        ch_multiqc_config
```

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf (hunk 3 above)
Comment (drpatelh) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf:28]:
```suggestion
ch_multiqc_config = file("$baseDir/assets/multiqc_config.yaml", checkIfExists: true)
```

Comment (drpatelh) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf:28]: Can change this back with the other change I made below.

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/nf-core/multiqc.nf [hunk 9]
```diff
@@ -5,35 +5,19 @@ if (!(workflow.runName ==~ /[a-z]+_[a-z]+/)) {
...
 /*
  * MultiQC
  */
 process MULTIQC {
     publishDir "${params.outdir}/multiqc", mode: params.publish_dir_mode
 
     input:
-    path (multiqc_config) from ch_multiqc_config
-    path (mqc_custom_config) from ch_multiqc_custom_config.collect().ifEmpty([])
+    path (multiqc_config)
```
Comment (drpatelh) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/nf-core/multiqc.nf:15]:
```suggestion
    path multiqc_config
```

Comment (drpatelh) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/nf-core/multiqc.nf:15]: Do we need the braces around all of these parameters?

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/nf-core/multiqc.nf [hunk 10]
```diff
@@ -46,6 +30,7 @@ process MULTIQC {
     custom_config_file = params.multiqc_config ? "--config $mqc_custom_config" : ''
     // TODO nf-core: Specify which MultiQC modules to use with -m for a faster run time
     """
-    multiqc -f $rtitle $rfilename $custom_config_file .
+    echo '$workflow_summary' > workflow_summary_mqc.yaml
+    multiqc -f $rtitle $rfilename $custom_config_file -m fastqc -m custom_content .
```
Comment (drpatelh) [inline nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/nf-core/multiqc.nf:34]:
```suggestion
    multiqc -f $rtitle $rfilename $custom_config_file .
```

Comment (drpatelh) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/nf-core/multiqc.nf:34]: Better to put the modules in `multiqc_config.yaml` e.g. [here](https://github.com/nf-core/viralrecon/blob/75a2f9763e9b4c1aa20ad342ae3ce148c33cc65a/assets/multiqc_config.yaml#L8-L22)

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf (hunk 3 above)
Comment (ggabernet) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/main.nf:28]: That also works, right. But using `file()` does not generate a channel but just a path. Then one cannot, e.g. print the content with `.dump`. Is this preferred? 
The following line `ch_multiqc_custom_config` also defines a channel with `Channel.fromPath()`. Maybe we can decide on which one is the preferred one and I'll change them all :)

File: nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/nf-core/multiqc.nf (hunk 9 above)
Comment (ggabernet) [reply nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/nf-core/multiqc.nf:15]: works fine without as well!

**Pull request merged**